app/javascript/components/widgets/widgets/forest-change/tree-loss/selectors.js

```diff
@@ -78,23 +79,33 @@ export const getSentence = createSelector(
     getIndicator,
     getSentences
   ],
-  (data, extent, settings, currentLabel, indicator, sentences) => {
+  (data, extent, settings, currentLocation, indicator, sentences) => {
     if (!data) return null;
-    const { initial, withIndicator, noLoss, noLossWithIndicator } = sentences;
+    const {
+      initial,
+      withIndicator,
+      noLoss,
+      noLossWithIndicator,
+      co2Emissions,
+      end
+    } = sentences;
     const { startYear, endYear, extentYear } = settings;
     const totalLoss = (data && data.length && sumBy(data, 'area')) || 0;
     const totalEmissions =
       (data && data.length && biomassToCO2(sumBy(data, 'emissions'))) || 0;
     const percentageLoss =
       (totalLoss && extent && totalLoss / extent * 100) || 0;
     let sentence = indicator ? withIndicator : initial;
+    sentence = tropicalIsos.includes(currentLocation.value)
+      ? sentence + co2Emissions
+      : sentence + end;
     if (totalLoss === 0) {
       sentence = indicator ? noLossWithIndicator : noLoss;
     }
 
     const params = {
       indicator: indicator && indicator.label.toLowerCase(),
-      location: currentLabel,
+      location: currentLocation.label,
```

Comment:
The reason you have been getting errors is because currentLocation is an object that when fetching data is sometimes not yet defined. You need to check it exists first.

Reply:
I think its this:

const getSettings = state => state.settings || null;

Reply:
Only if you are accessing `settings.something` without checking. Either way you need to check you have those other objects defined.